Zend/zend_compile.c

```diff
@@ -5904,15 +5882,13 @@ void zend_compile_prop_decl(zend_ast *ast, zend_ast *type_ast, uint32_t flags) /
 			zend_const_expr_to_zval(&value_zv, value_ast);
 
 			if (ZEND_TYPE_IS_SET(type) && !Z_CONSTANT(value_zv)) {
-				if (Z_TYPE(value_zv) == IS_NULL) {
-					if (!ZEND_TYPE_ALLOW_NULL(type)) {
-						const char *name = ZEND_TYPE_IS_CLASS(type)
-							? ZSTR_VAL(ZEND_TYPE_NAME(type)) : zend_get_type_by_const(ZEND_TYPE_CODE(type));
-						zend_error_noreturn(E_COMPILE_ERROR,
-								"Default value for property of type %s may not be null. "
-								"Use the nullable type ?%s to allow null default value",
-								name, name);
-					}
+				if (Z_TYPE(value_zv) == IS_NULL && !ZEND_TYPE_ALLOW_NULL(type)) {
```

Comment:
This changes the logic of the code, as now the value == null && type allow null && type is class case generates an error, while it shouldn't.